File: Python/3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_len=0
        left=0
        chars=set()
        for right in range(len(s)):
            #print(chars)
            while s[right] in chars:
                chars.remove(s[left])
                left+=1
            chars.add(s[right])
            current_len=right-left+1
            max_len=max(max_len,current_len)
        return max_len
      

File: Java/3. Longest Substring Without Repeating Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l = 0;
        int res = 0;
        HashSet<Character> set = new HashSet<>();
        for (int r=0;r<s.length();r++){
            char ch = s.charAt(r);
            while (set.contains(ch)){
                set.remove(s.charAt(l));
                l++;
            }
            set.add(s.charAt(r));
            res = Math.max(res,r-l+1);
        }
        return res;
    }
}